File: uoj_1368_dna-consensus-string.py
cases = int(input())

while cases:
    # m: number of sequences
    # n: length of sequences
    m,n = [int(i) for i in input().split()]

    # Store the sequences
    sequences = []
    for i in range(m):
        sequences.append(input())

    consensusString = []
    consensusError = 0

    # The algorithm I'm thinking of (maybe too simplistic), is O(mxn)
    # For every character
    for index in range(n):
        # Create a map for all possible characters: ACTG
        characterMap = {"A":0, "C": 0, "G":0, "T":0}
        for seq in range(m):
            characterMap[sequences[seq][index]] += 1
            #increase the corresponding map

        # find the character with the gratest number of iterations, if
        # there are two of them, alphabetically (thus, use the map in
        # alphabetic order)
        consensusLetter = "A"
        error = 0
        for letter in ["T","C","G"]:
            if characterMap[letter] > characterMap[consensusLetter]:
                error += characterMap[consensusLetter]
                consensusLetter = letter
            elif characterMap[letter]:
                error += characterMap[letter]
        consensusString.append(consensusLetter)
        consensusError += error 

    print("".join((consensusString)))
    print(consensusError)

    cases -= 1

File: uoj_1368_dna-consensus-string.cpp
#include <string>
#include <iostream>
#include <map>

using namespace std;


char myArray[] = {'C','G','T'};
char letter;

int main()
{
  int cases;
  int m, n;

  cin >> cases;
  while (cases--)
    {
      cin >> m >> n;
      
      string sequences[m];
      getline(cin,sequences[0]); // Ignore \n

      for (int index = 0; index < m; index++)
	{
	  getline(cin, sequences[index]);
	  // cout << "\t>" << sequences[index] << endl;
	}

      string consensusString = sequences[0];
      long consensusError = 0;
      char consensusLetter = 'A';


      for (int index = 0; index < n; index++)
	{
	  map<char,int> characterMap ;
	  characterMap['A'] = 0;
	  characterMap['C'] = 0;
	  characterMap['G'] = 0;
	  characterMap['T'] = 0;

	  for (int seq = 0; seq < m; seq++)
            characterMap[sequences[seq][index]]++;

	  consensusLetter = 'A';
	  for (int i = 0; i < 3; i++)
	    {
	      letter = myArray[i];
	      if (characterMap[letter] > characterMap[consensusLetter])
		consensusLetter = letter;
	    }

	  consensusString[index] = consensusLetter;
	  consensusError += m - characterMap[consensusLetter];
	}
      // cout << "\t>" << m << " " << n << endl;
      cout << consensusString << endl;
      cout << consensusError << endl;
    }
  
}
